### PR TITLE
MAXIV: update Machine Info HWO

### PR DESCRIPTION
**Pull request by elmjag**
Updated MachInfo hardware object, for use at MAXIV BioMAX and MicroMAX beamlines.

Based on older BioMAX HWO, updated to follow the new AbstractMachineInfo API and refactored.

### CODE REVIEW

**Review by beteva — COMMENTED** [review]
Hi @elmjag, nice to have provided tests.  
This is your specific HO, but I was wondering, as it calls tango devices, is there a reason you do not use TangoMachineInfo?

**Comment by elmjag** [comment]
Not sure, this implementation is based on old version of MachineInfo HWO from BioMAX.

I did not knew that TangoMachineInfo existed. I'll take a look if we can use that instead.

**Comment by elmjag** [comment]
I have looked at TangoMachineInfo now. The TangoMachineInfo is designed to read all machine info topics from one specified tango device.

Here at MAXIV the topics are provided by multiple tango devices. We also need to do some parsing for a topic, thus we'll need to have a custom MachineInfo HWO anyway.

I feel that trying to use TangoMachineInfo in our case will make the implementation more complicated compared to how it's done in this PR.

File: mxcubecore/HardwareObjects/MAXIV/MachInfo.py [hunk 1]
```diff
@@ -1,153 +1,109 @@
...
 
-   optional fields:
-      any number of optional fields can be sent over with this signal by adding them in the
-      values dictionary
 
-      for example:
-         values['lifetime']
-         values['topup_remaining']
-"""
+def catch_errs(func):
```
Comment (marcus-oscarsson) [inline mxcubecore/HardwareObjects/MAXIV/MachInfo.py]: :), I would have simply called it `catch_errors` but its  matter of taste I guess and it is after all site specific code

File: mxcubecore/HardwareObjects/MAXIV/MachInfo.py [hunk 2]
```diff
@@ -1,153 +1,109 @@
...
+import gevent
+from tango import DeviceProxy
+from mxcubecore.utils.units import sec_to_hour, A_to_mA
+from mxcubecore.HardwareObjects.abstract.AbstractMachineInfo import AbstractMachineInfo
 
-[Description]
-MachInfo hardware objects are used to obtain information from the accelerator
-control system.
+# how often we refresh machine info
+REFRESH_PERIOD_SEC = 30
```
Comment (marcus-oscarsson) [inline mxcubecore/HardwareObjects/MAXIV/MachInfo.py:10]: You could also easily make it configurable via `get_property`  but perhaps not desired in your case.

**Review by marcus-oscarsson — APPROVED** [review]

**Comment by marcus-oscarsson** [comment]
Looks good to me, Ill let you rebase and merge :), you (of-course) do as you like with the comments

File: mxcubecore/HardwareObjects/MAXIV/MachInfo.py (hunk 1 above)
Comment (elmjag) [reply mxcubecore/HardwareObjects/MAXIV/MachInfo.py]: good point, I have changed it to `catch_errors`

File: mxcubecore/HardwareObjects/MAXIV/MachInfo.py (hunk 2 above)
Comment (elmjag) [reply mxcubecore/HardwareObjects/MAXIV/MachInfo.py:10]: I feel it's not worth the hassle in this case. We have not changed this value on a very long time.

**Comment by elmjag** [comment]
> Looks good to me, Ill let you rebase and merge :), you (of-course) do as you like with the comments

I have re-based, but I don't have the commit bit to do the merge. :disappointed:

**Comment by marcus-oscarsson** [comment]
Oh, I think it should be ok now :)

**Pull request merged**